encrypted_keys.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 The age Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package nknwallet
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"filippo.io/age"
+)
+
+// LazyScryptIdentity is an age.Identity that requests a passphrase only if it
+// encounters an scrypt stanza. After obtaining a passphrase, it delegates to
+// ScryptIdentity.
+type LazyScryptIdentity struct {
+	Passphrase func() (string, error)
+}
+
+var _ age.Identity = &LazyScryptIdentity{}
+
+func (i *LazyScryptIdentity) Unwrap(stanzas []*age.Stanza) (fileKey []byte, err error) {
+	for _, s := range stanzas {
```

Comment:
Is this for loop redundant? Seems it's completely covered by L30-32

Reply:
fixed.